SEED-platform/seed
# Improve data logger validation error messages

#### Any background context you want to provide?
The data logger create/update modal had some hard coded error messages and backend logic that didn't align.

#### What's this PR do?
Dynamically displays error messages to the user 
Refactors the create and update endpoints to use a serializer to validate data and generate errors.

Based on the existing data logger model, the only required fields are property, display_name, and is_occupied_data. I used this as the source of truth for refactoring the view logic.

#### How should this be manually tested?
Create/update a data logger, make sure error messages are applicable.
- testing scenarios: name is empty, name is not unique. Otherwise, all other required fields are assigned in the backend

#### What are the relevant tickets?
#4592

#### Screenshots (if appropriate)
